pkg/alertmanager/api_test.go

```diff
@@ -0,0 +1,74 @@
+package alertmanager
+
+import (
+	"testing"
+
+	"github.com/cortexproject/cortex/pkg/alertmanager/alerts"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestAMConfigValidation(t *testing.T) {
```

Comment:
One last nit (but maybe make sure we create an issue for it?) - I'm a big fan of tests talk to the handler directly over using [`httptest`](https://golang.org/pkg/net/http/httptest/#NewRequest). They can help quickly illustrate what the API expects and returns in terms of format.

I wouldn't mind if we just create the base structure with 1 fail / 1 pass but serves as a baseline to extend with other cases as we make changes here.